### PR TITLE
[candi] Remove early-oom

### PR DESCRIPTION
**Pull request by Diaderius**
## Description

Removed early-oom. Added cgroup memory reservation.

## Why do we need it, and what problem does it solve?

We'd like to prevent nodes from getting stuck on memory pressure. Unfortunately, early-oom was a poor design decision, since PSI metrics do not differentiate between cgroup-level memory pressure and global memory pressure. It turns out, that kubelet memory reservation options put a memory limit on the whole `kubepods.slice`, effectively preventing Node from getting stuck on memory page thrashing.

## What is the expected result?
Early-oom would be disabled at all.

1. Nodes are not getting stuck on page thrashing.
2. Manual OOM does not randomly trigger on cgroup-contained memory pressure events

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: node-manager
type: fix
summary: Removed early-oom. Added kubelet memory reservation option.
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Diaderius. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Diaderius. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by zuzzas** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.23 4355414264 4355414264-1-con-1-23 /sys/deckhouse-oss/install:pr3821 cloud-user@158.160.58.176

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by apolovov** [comment]
It is more about [node-manager] imo.

**Comment by zuzzas** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.23 4372013294 4372013294-1-con-1-23 /sys/deckhouse-oss/install:pr3821 cloud-user@158.160.60.240

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: candi/bashible/common-steps/node-group/064_configure_kubelet.sh.tpl [hunk 1]
```diff
@@ -151,6 +151,74 @@ if [[ -f /var/lib/bashible/cloud-provider-variables ]]; then
   fi
 fi
 
+function dynamic_memory_sizing {
```
Comment (RomanenkoDenys) [inline candi/bashible/common-steps/node-group/064_configure_kubelet.sh.tpl:154]: need comment where this function is.

File: ee/modules/040-node-manager/openapi/config-values.yaml [hunk 2]
```diff
@@ -1,12 +1,5 @@
 type: object
 properties:
-  earlyOomEnabled:
-    type: boolean
-    default: true
-    x-examples:
-    - false
-    description: |
-      Set to 'false' to disable early OOM killer in case it behaves incorrectly.
```
Comment (RomanenkoDenys) [inline ee/modules/040-node-manager/openapi/config-values.yaml:9]: We need module config conversion here

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
nits

File: candi/bashible/common-steps/node-group/064_configure_kubelet.sh.tpl (hunk 1 above)
Comment (zuzzas) [reply candi/bashible/common-steps/node-group/064_configure_kubelet.sh.tpl:155]: Done.

File: ee/modules/040-node-manager/openapi/config-values.yaml (hunk 2 above)
Comment (zuzzas) [reply ee/modules/040-node-manager/openapi/config-values.yaml:9]: Done.

File: candi/bashible/common-steps/node-group/064_configure_kubelet.sh.tpl (hunk 1 above)
Comment (zuzzas) [reply candi/bashible/common-steps/node-group/064_configure_kubelet.sh.tpl:155]: Done.

File: modules/040-node-manager/settings-conversion/conversion.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2022 Flant JSC
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/settings-conversion/conversion.go:2]:
```suggestion
Copyright 2023 Flant JSC
```

File: modules/040-node-manager/settings-conversion/conversion_test.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2022 Flant JSC
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/settings-conversion/conversion_test.go:2]:
```suggestion
Copyright 2023 Flant JSC
```

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]

**Review by yalosev — CHANGES_REQUESTED** [review]
Not completed

**Review by RomanenkoDenys — APPROVED** [review]

**Review by yalosev — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Pull request merged**